api/controllers/build.js

```diff
@@ -134,6 +135,17 @@ module.exports = wrapHandlers({
       EventCreator.error(Event.labels.BUILD_STATUS, err, { buildId: build.id });
     }
 
+    // we start certain build tasks on build completion
+    if (body.status === 'success') {
+      try {
+        const tasks = await SiteBuildTask
```

Comment:
What's your thoughts on model instance hooks? We could add the `startBuildTasks` on the `afterSave` or `afterUpdate` hook and then have a higher order function that starts the proper build tasks based on the updated `status`. https://sequelize.org/docs/v6/other-topics/hooks/#instance-hooks

Does it simplify starting the build tasks? Is it too coupled to the ORM?

Reply:
I think this one would be tricky because builds are kicked off via `.enqueue` which always follows `Build.create` in our codebase but the two aren't guaranteed to be linked. Until we re-investigate some of the status/queue logic, I'm a little worried about skipping/double-starting build tasks if we link it to the instance hooks (even though it is probably a "cleaner" method overall

Reply:
I tried this and it seemed to work fine; it was nice to remove the logic from the build controller.